### PR TITLE
Allow removing units if minAlive=0

### PR DESCRIPTION
**Pull request by Nesurion**
In case the min-alive value is set to 0, we allow removing units.

### CODE REVIEW

**Comment by Nesurion** [comment]
RFR @giantswarm/cena

File: controller/update.go [hunk 1]
```diff
@@ -79,8 +79,9 @@ func (c controller) isGroupRemovalAllowed(ctx context.Context, req Request, minA
 		ctx, "controller: removeInProgress: %v, minAlive: %v",
 		*removeInProgress, minAlive,
 	)
-
-	if (minAlive - int(*removeInProgress)) > 0 {
+	// if minAlive = 0 we don't need to calculate if we can remove slices,
+	// as the user provided us with the information, that killing slices is fine
```
Comment (xh3b4sd) [inline controller/update.go:83]: `that killing ALL slices`

File: controller/update.go [hunk 2]
```diff
@@ -79,8 +79,9 @@ func (c controller) isGroupRemovalAllowed(ctx context.Context, req Request, minA
...
+	if (minAlive-int(*removeInProgress)) > 0 || minAlive == 0 {
 		c.Config.Logger.Debug(ctx, "controller: group removal allowed ((minAlive - int(removeInProgress)) > 0)")
```
Comment (xh3b4sd) [inline controller/update.go]: We may want to adapt the logging accordingly.

File: int-tests/007-min-alive-zero-update.t [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  $ inagoctl --fleet-endpoint=${FLEET_ENDPOINT} start $GROUP > /dev/null 2>&1
+  $ sleep 3
+
+Modify unit
+
+  $ echo "[Unit]\nDescription=Unit 1 (CHANGED)\n[Service]\nExecStart=/bin/bash -c 'while true; do echo Hello %n; sleep 10; done'\n" > $GROUP/$GROUP-unit@.service
+
+Update unit
+
+  $ inagoctl --fleet-endpoint=${FLEET_ENDPOINT} update $GROUP --max-growth=0 --min-alive=0
```
Comment (xh3b4sd) [inline int-tests/007-min-alive-zero-update.t]: IIRC we discussed turning the ready secs down so the int tests are running faster. Why don't we do this?

**Comment by xh3b4sd** [comment]
Why are codecov results going crazy?

**Comment by Nesurion** [comment]
But in the `ci.sh` script prevents travis from stopping the build if the unit tests fail.
See https://github.com/giantswarm/inago/pull/246

File: int-tests/007-min-alive-zero-update.t (hunk 3 above)
Comment (Nesurion) [inline int-tests/007-min-alive-zero-update.t]: we discussed here: https://github.com/giantswarm/inago/pull/233
We didn't investigate further yet why the tests tend to fail without the sleep in there. Will be done in a seperate PR for all int tests.

Comment (xh3b4sd) [inline int-tests/007-min-alive-zero-update.t]: As far as I understood this was about the `sleep`. I am talking about `--ready-secs`. It should be possible to set this down to 5 seconds.

**Comment by Nesurion** [comment]
RFR @giantswarm/cena

File: int-tests/007-min-alive-zero-update.t [hunk 4]
```diff
@@ -0,0 +1,32 @@
+This test checks the behaviourof an update with max-growth zero
```
Comment (JosephSalisbury) [inline int-tests/007-min-alive-zero-update.t]: behaviour of

**Comment by JosephSalisbury** [comment]
yep, LGTM!

File: int-tests/007-min-alive-zero-update.t [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  $ inagoctl --fleet-endpoint=${FLEET_ENDPOINT} start $GROUP > /dev/null 2>&1
+  $ sleep 3
+
+Modify unit
+
+  $ echo "[Unit]\nDescription=Unit 1 (CHANGED)\n[Service]\nExecStart=/bin/bash -c 'while true; do echo Hello %n; sleep 10; done'\n" > $GROUP/$GROUP-unit@.service
+
+Update unit
+
+  $ inagoctl --fleet-endpoint=${FLEET_ENDPOINT} update $GROUP --max-growth=0 --min-alive=0 --ready-secs=5
```
Comment (xh3b4sd) [inline int-tests/007-min-alive-zero-update.t]: Why ready secs 5 ? IIRC I recently saw another test where it was 3 or 2.

Comment (Nesurion) [inline int-tests/007-min-alive-zero-update.t]: There is currently no other tests that sets the `ready-secs` flag to something other than the default (30).
Setting it to 5 is arbitrary value less than 30. No particular reason, more a gut feeling.

Comment (xh3b4sd) [inline int-tests/007-min-alive-zero-update.t]: Good I don't trust anybody. https://github.com/giantswarm/inago/commit/45ea56562f031bde35ac18847d8664e16c8afc3f#diff-3c62d059684b69b3b81228eb8f747593R22

**Comment by Nesurion** [comment]
@xh3b4sd RFR

**Comment by xh3b4sd** [comment]
LGTM

**Pull request merged**